core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java

```diff
@@ -99,8 +99,8 @@ public class NativeImageFeatureStep {
     static final String BEFORE_ANALYSIS_ACCESS = Feature.BeforeAnalysisAccess.class.getName();
     static final String DURING_SETUP_ACCESS = Feature.DuringSetupAccess.class.getName();
     static final String DYNAMIC_PROXY_REGISTRY = "com.oracle.svm.core.jdk.proxy.DynamicProxyRegistry";
-    static final String LEGACY_LOCALIZATION_FEATURE = "com.oracle.svm.core.jdk.LocalizationFeature";
     static final String LOCALIZATION_FEATURE = "com.oracle.svm.core.jdk.localization.LocalizationFeature";
+    static final String RUNTIME_RESOURCE_SUPPORT = "org.graalvm.nativeimage.impl.RuntimeResourceSupport";
```

Comment:
To my understanding they are both public APIs that will be shipped in graal-sdk. The reason I went with `RuntimeResourceSupport` instead of `RuntimeResourceAccess` is that the former doesn't require to specify a module for the resources, and in Quarkus we are not using modules.

`org.graalvm.nativeimage.hosted.RuntimeResourceAccess` is part of the `org.graalvm.nativeimage.hosted` package and `org.graalvm.nativeimage.impl.RuntimeResourceSupport` is part of the `org.graalvm.nativeimage` package which are [both included in graal-sdk](https://github.com/oracle/graal/blob/3fa9fb1b71abc7bd75e4bf2e2f81fdb281cc2926/sdk/mx.sdk/suite.py#L422-L428).

Reply:
OK. When I see `impl` in an API class it raises a red flag to me. While it's true `RuntimeResourceSupport` doesn't require a module, we should get away with `ClassLoader.getSystemClassLoader().getUnnamedModule()` and use the `RuntimeResourceAccess`. In fact, `RuntimeResourceAccess` uses `RuntimeResourceSupport` which supports the idea that using the former will be a more robust solution. Also, you don't need to do the `ImageSingleton` handling yourself and it's in line with what we do in https://github.com/quarkusio/quarkus/pull/27690. Your mileage may vary.

Reply:
The tough part is not about getting the unnamed module but getting the named module for JDK classes, e.g. https://github.com/quarkusio/quarkus/blob/1f87a535ed1a053e45ba56fd04aced9d0da041cd/core/deployment/src/main/java/io/quarkus/deployment/steps/ResourceBundleStep.java#L15. Suggestions are welcome :)

Reply:
`sun.security.util.Resources` is part of `java.base` so how about? `int.class.getModule()`

Reply:
This is just an example, there are more such cases and while getting the module might not be that hard what makes it hard is the fact that we need to get the module in the generated native-image `Feature`. So to make use of the `RuntimeResourceAccess` we need to:

1. Extend `NativeImageResourceBundleBuildItem` to contain information about the module
2. Generate the necessary code, using `gizmo`, to programmatically get that module and pass it to `RuntimeResourceAccess` in the generated native-image `Feature`.

Since `RuntimeResourceSupport` appears to be public API I don't see the need to do the above which will make the corresponding code even more complex.

As an alternative I believe that if we really don't want to use `RuntimeResourceSupport` we should suggest the augmentation of `RuntimeResourceAccess` to support registering resources without the module name.

Reply:
Apparently, I was wrong. `org.graalvm.nativeimage.impl` is indeed not supposed to be public API. But we already open it in https://github.com/quarkusio/quarkus/blob/92fba0ab960b5cdb432e87f53e16fcfff106bac5/core/deployment/src/main/java/io/quarkus/deployment/steps/NativeImageFeatureStep.java#L130-L132 which led me to thinking it's accessible by default.

I will need to prepare a follow up fix for this. Thanks for questioning this and making have another look @jerboaa!

Reply:
@zakkak Thanks for coming back to this! I was meaning to come back to this discussion, but it dropped on the floor (for me). Yes, we should revisit this. IMHO, there might be latent bugs by not passing in Modules resources belong to.

Reply:
WIP PR taking care of the above: https://github.com/quarkusio/quarkus/pull/28093